### PR TITLE
Backend state changes

### PR DESCRIPTION
**Pull request by Josh-Matsuoka**
Hi,

This PR addresses https://github.com/cryostatio/cryostat/issues/354 , sending events on the Notification channel when recordings are created/deleted, archived, and templates are uploaded/deleted. I've adjusted the tests accordingly.

### CODE REVIEW

File: src/main/java/io/cryostat/net/web/http/api/v1/RecordingDeleteHandler.java [hunk 1]
```diff
@@ -110,6 +117,13 @@ public void handleAuthenticated(RoutingContext ctx) throws Exception {
                                 throw new HttpStatusException(500, e.getMessage(), e);
                             } finally {
                                 reportService.delete(recordingName);
+                                notificationFactory
+                                        .createBuilder()
+                                        .metaCategory(NOTIFICATION_CATEGORY)
+                                        .metaType(HttpMimeType.JSON)
+                                        .message(Map.of("recording", recordingName))
+                                        .build()
+                                        .send();
```
Comment (andrewazores) [inline src/main/java/io/cryostat/net/web/http/api/v1/RecordingDeleteHandler.java:126]: I think this should be a few lines higher up in the main body of the `try` - if the deletion fails and an exception is thrown, then a notification should probably not be emitted.

**Comment by andrewazores** [comment]
Looks good otherwise!

File: src/main/java/io/cryostat/net/web/http/api/v1/RecordingsPostHandler.java [hunk 2]
```diff
@@ -81,6 +82,8 @@
     private final Path savedRecordingsPath;
     private final Gson gson;
     private final Logger logger;
+    private final NotificationFactory notificationFactory;
+    private static final String NOTIFICATION_CATEGORY = "RecordingArchived";
```
Comment (andrewazores) [inline src/main/java/io/cryostat/net/web/http/api/v1/RecordingsPostHandler.java:86]: Oh, I think this category should be renamed - `RecordingArchived` sounds like an in-memory/active recording was copied into archives, but this handler allows users to re-upload previously archived recordings from their workstation into Cryostat.

File: src/main/java/io/cryostat/net/web/http/api/v1/TargetRecordingDeleteHandler.java [hunk 3]
```diff
@@ -103,6 +110,13 @@ public void handleAuthenticated(RoutingContext ctx) throws Exception {
                     if (descriptor.isPresent()) {
                         connection.getService().close(descriptor.get());
                         reportService.delete(connectionDescriptor, recordingName);
+                        notificationFactory
+                                .createBuilder()
+                                .metaCategory(NOTIFICATION_CATEGORY)
+                                .metaType(HttpMimeType.JSON)
+                                .message(Map.of("recording", recordingName))
```
Comment (andrewazores) [inline src/main/java/io/cryostat/net/web/http/api/v1/TargetRecordingDeleteHandler.java:117]: Maybe this payload should also include the `targetId` (really a JMX service URL) of the target that "owns" the named recording?

Comment (Josh-Matsuoka) [reply src/main/java/io/cryostat/net/web/http/api/v1/TargetRecordingDeleteHandler.java]: Makes sense, I've added this to the target-oriented handlers' payloads

File: src/main/java/io/cryostat/net/web/http/api/v1/RecordingsPostHandler.java (hunk 2 above)
Comment (Josh-Matsuoka) [reply src/main/java/io/cryostat/net/web/http/api/v1/RecordingsPostHandler.java]: makes sense, I've renamed this now

File: src/main/java/io/cryostat/net/web/http/api/v1/RecordingDeleteHandler.java (hunk 1 above)
Comment (Josh-Matsuoka) [reply src/main/java/io/cryostat/net/web/http/api/v1/RecordingDeleteHandler.java:122]: makes sense, I moved this and adjusted the test accordingly.

**Review by andrewazores — APPROVED** [review]

**Comment by andrewazores** [comment]
@Josh-Matsuoka do you have any interest in looking at https://github.com/cryostatio/cryostat-web/issues/165 ? That's the corresponding frontend issue. This would simply entail adding notification handlers for each of these new notification types emitted and either using the notification payload to update the view, or if that's infeasible then to use the notification as a trigger to poll for fresh data.

**Comment by Josh-Matsuoka** [comment]
I can take a look at that, sure

**Pull request merged**